connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceProvider.scala

```diff
@@ -608,6 +613,59 @@ private[kafka010] object KafkaSourceProvider extends Logging {
   private val serClassName = classOf[ByteArraySerializer].getName
   private val deserClassName = classOf[ByteArrayDeserializer].getName
 
+  def checkStartOffsetNotGreaterThanEndOffset(
+      startOffset: Long,
+      endOffset: Long,
+      topicPartition: TopicPartition,
+      exception: (Long, Long, TopicPartition) => Exception): Unit = {
+    if (startOffset > endOffset && startOffset != -1 && endOffset != -1) {
```

Comment:
Same here, why -1 is handled separately here? Shall we leave a code comment since this looks like a magic number for a random reader?